File: apps/ai-service/src/routes/analyze.ts
import { Router } from "express";

import { getProvider } from "../providers/index.js";
import { requireInternalSecret } from "../security.js";
import { AnalyzeReportRequestSchema } from "../types.js";

export const analyzeRouter = Router();

analyzeRouter.post("/analyze", requireInternalSecret, async (req, res) => {
  const parsed = AnalyzeReportRequestSchema.safeParse(req.body);
  if (!parsed.success) {
    res.status(400).json({ error: "Invalid request", details: parsed.error.flatten() });
    return;
  }

  try {
    const result = await getProvider().analyze(parsed.data);
    res.json(result);
  } catch (err) {
    res.status(502).json({ error: "AI provider failed", detail: String(err) });
  }
});

File: apps/ai-service/app/routes/analyze.py
from fastapi import APIRouter, Depends

from app.models.schemas import AnalyzeReportRequest, AnalyzeReportResult
from app.providers import get_provider
from app.security import require_internal_secret

router = APIRouter()


@router.post("/analyze", response_model=AnalyzeReportResult)
async def analyze(
    request: AnalyzeReportRequest,
    _: None = Depends(require_internal_secret),
) -> AnalyzeReportResult:
    provider = get_provider()
    return await provider.analyze(request)
